src/middlewared/middlewared/plugins/smb.py

```diff
@@ -494,6 +495,42 @@ async def ctdb_wait(self):
 
             await asyncio.sleep(1)
 
+    @private
+    async def netif_wait(self, timeout=120):
+        """
+        Wait for for the ix-netif sentinel file
+        and confirm connectivity with some targeted tests.
+        All must be completed before the timeout.
+        """
+        found_sentinel = False
+        while timeout >= 0 and not found_sentinel:
+            if os.path.exists(NETIF_COMPLETE_SENTINEL):
+                os.remove(NETIF_COMPLETE_SENTINEL)
+                found_sentinel = True
+
+            timeout -= 1
+            if timeout <= 0:
+                self.logger.warning('Failed to get netif completion sentinal.')
+            elif not found_sentinel:
+                await asyncio.sleep(1)
+
+        """
+        Confirm at least one network interface is UP
+        """
+        if_UP = False
+        while timeout >= 0 and not if_UP:
+            netif = await self.middleware.call('interface.query')
```

Comment:
There is no reason to generate lists of interface names when we just need to detect if any of them are up. I suggest doing something like this
```suggestion
    while timeout >= 0:
        if any((
            i['state']['link_state'] == 'LINK_STATE_UP' for i in await self.middleware.call('interface.query')
        )):
            break
        else:
            timeout -= 1
            if timeout <= 0:
                self.logger.warning('Failed to detect any connected network interfaces.')
            else:
                await asyncio.sleep(1)
```

Reply:
👍